tests/integration/component.go

```diff
@@ -693,4 +694,52 @@ func componentTests(args ...string) {
 			Expect(stdError).To(ContainSubstring("quantities must match the regular expression"))
 		})
 	})
+
+	Context("Creating component using symlink", func() {
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+			originalDir = helper.Getwd()
+			helper.Chdir(context)
+		})
+		JustAfterEach(func() {
+			// remove the symlink
+			err := os.Remove(symLinkPath)
+			Expect(err).NotTo(HaveOccurred())
+
+			helper.DeleteProject(project)
+			helper.Chdir(originalDir)
+		})
+
+		It("Should be able to deploy a spring boot uberjar file using openjdk", func() {
+			oc.ImportJavaIS(project)
+
+			// change to the parent of the context
+			helper.Chdir(filepath.Dir(context))
+
+			// create a symlink
+			symLinkName := "symLink"
+			helper.SymLink(filepath.Base(context), symLinkName)
```

Comment:
Shouldn't the symlink creation be part of the test setup (in `JustBeforeEach`)?

Reply:
Wouldn't this create a symlink to context dir inside the context dir itself?

if the context is for example `/tmp/test` the symlink will be `/tmp/test/symLink` pointing to `/tmp/test`, is that what you want, or am I missing something?

Reply:
Also, this doesn't look right at all. For example, if the context is `/tmp/test/1` this will call `symlink("1","symLink") in the `/tmp/test/1` directory and it will fail.

Reply:
It would be better and cleaner to create a new tempory dir, and create symlink inside it

Reply:
I did it because someone might want to add a similar test in the same context which will have a symlink to a file

Reply:
> Also, this doesn't look right at all. For example, if the context is /tmp/test/1 this will call symlink("1","symLink") in the /tmp/test/1` directory and it will fail.

I am switching to the parent directory which is `/tmp` before running the test and thus the symlink is created inside `tmp`

Reply:
> I am switching to the parent directory which is `/tmp` before running the test and thus the symlink is created inside `tmp`

I missed that. But still this is not good. As you are switching from a randomly created directory to on elevel up. If two such tests are run at the same time it will fail on as they will both they to create the same symlink.
```
    Unexpected error:
        <*os.LinkError | 0xc0002d8400>: {Op: "symlink", Old: "145303116", New: "symLink", Err: 0x11}
        symlink 145303116 symLink: file exists
    occurred
```

Reply:
> I did it because someone might want to add a similar test in the same context which will have a symlink to a file

if we need such a test we can always split it into nested context and break `JustBeforeEach` and `JustAfterEach` into two